pkg/apis/core/validation/validation.go

```diff
@@ -5360,9 +5360,33 @@ func ValidateSecurityContext(sc *core.SecurityContext, fldPath *field.Path) fiel
 // maxGMSACredentialSpecLength is the max length, in bytes, for the actual contents
 // of a GMSA cred spec. In general, those shouldn't be more than a few hundred bytes,
 // so we want to give plenty of room here while still providing an upper bound.
+// The runAsUserName field will be used to execute the given container's entrypoint, and
+// it can be formatted as "DOMAIN/USER", where the DOMAIN is optional, maxRunAsUserNameDomainLength
+// is the max character length for the user's DOMAIN, and maxRunAsUserNameUserLength
+// is the max character length for the USER itself. Both the DOMAIN and USER have their
+// own restrictions, and more information about them can be found here:
+// https://support.microsoft.com/en-us/help/909264/naming-conventions-in-active-directory-for-computers-domains-sites-and
+//  https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.localaccounts/new-localuser?view=powershell-5.1
 const (
 	maxGMSACredentialSpecLengthInKiB = 64
 	maxGMSACredentialSpecLength      = maxGMSACredentialSpecLengthInKiB * 1024
+	maxRunAsUserNameDomainLength     = 256
+	maxRunAsUserNameUserLength       = 21
+)
+
+var (
+	// a valid NetBios Domain name cannot start with a dot, has at least 1 character,
+	// at most 15 characters, and it cannot the characters: \ / : * ? " < > |
+	// we're also excluding the control characters (ASCII codes: 0-31, 127; in hexa: 00-1F, 7F)
+	validNetBiosRegex, _ = regexp.Compile(`^(?:[^\x00-\x1F\x7F\\/:\*\?"<>|\.][^\x00-\x1F\x7F\\/:\*\?"<>|]{0,14})?$`)
```

Comment:
nit: can use `[:cntrl:]` in place of `\x00-\x1F\x7F` for slightly better readability. up to you if you want to use that

Reply:
the final `?` allows validNetBiosRegex to match empty strings, right? that doesn't seem correct